src/ThemeProvider.js

```diff
@@ -0,0 +1,89 @@
+import React, {PropTypes, Component} from 'react'
+import mitt from 'mitt'
+
+export const CHANNEL = '__glamorous__'
+
+/**
+ * This is a component which will provide a theme to the entire tree
+ * via context and event listener (because pure components block conext updates)
+ * @param {Object} props the props for the component.
+ * @param {Object} context the context for the component.
+ * @return {ReactElement} React.createElement
+ */
+class ThemeProvider extends Component {
+  constructor(props, context) {
+    super(props, context)
+    this.getTheme = this.getTheme.bind(this)
+    this.setOuterTheme = this.setOuterTheme.bind(this)
+    this.emitter = mitt()
+    this.publish = this.emitter.emit.bind(this, CHANNEL)
+    this.subscribe = this.subscribe.bind(this)
+  }
+
+  // subscribe to future theme changes, emitting once to setup the theme,
+  // and returning the unsubscriptio functon
+  subscribe(subscription) {
+    const emitter = this.emitter
+    emitter.on(CHANNEL, subscription)
+    emitter.emit(CHANNEL, this.getTheme())
+
+    return function unsubscribe() {
+      emitter.off(CHANNEL, subscription)
+    }
+  }
+
+  // create them, by merging with outer theme, if present
+  getTheme(passedTheme) {
+    const theme = passedTheme || this.props.theme
+    return {...this.outerTheme, ...theme}
+  }
+
+  getChildContext() {
+    return {
+      ...this.context,
+      [CHANNEL]: {
```

Comment:
I do not know. Personally I would not give this freedom to users.
To me a theme is something global and I've never felt the need of having more than one in same app. (but maybe I've never ran into that scenario 😱)

Do you think it would be a nice addition considering we already provide merging capabilities? 
Should not be complex to implement.